src/Data/Vec/Relation/Binary/Lex/NonStrict.agda

```diff
@@ -0,0 +1,196 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Lexicographic ordering of vector
+------------------------------------------------------------------------
+
+-- The definitions of lexicographic orderings used here is suitable if
+-- the argument order is a (non-strict) partial order.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Relation.Binary.Lex.NonStrict where
+
+open import Data.Empty
+open import Data.Unit using (⊤; tt)
+open import Data.Product using (proj₁; proj₂)
+open import Data.Nat using (ℕ)
+open import Data.Vec using (Vec; []; _∷_)
+import Data.Vec.Relation.Binary.Lex.Strict as Strict
+open import Data.Vec.Relation.Binary.Pointwise.Inductive as Pointwise using (Pointwise; []; _∷_; head; tail)
+open import Function using (id)
+open import Relation.Binary
+import Relation.Binary.Construct.NonStrictToStrict as Conv
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+open import Relation.Nullary hiding (Irrelevant)
+open import Level using (_⊔_)
+
+------------------------------------------------------------------------
+-- Publicly re-export definitions from Core
+
+open import Data.Vec.Relation.Binary.Lex.Core as Core public
+  using (base; this; next; ¬≤-this; ¬≤-next)
+
+----------------------------------------------------------------------
+-- Strict lexicographic ordering.
+
+module _ {a ℓ₁ ℓ₂} {A : Set a} where
```

Comment:
I would make `{a ℓ₁ ℓ₂}` into variables and you can save an indentation level.

Reply:
I don't save the indent level without `{A : Set a}` and for moving everything into variable there seems to be distinction between `A.a` and `a`:

```agda
variable
  a ℓ₁ ℓ₂ : Level
  A : Set a
```

Reply:
Yup, when you have `A : Set a` as a variable, you'll still need to put the `{A : Set a}` into the definition as follows:
```agda
data Lex {A : Set a} (P : Set) (_≈_ : Rel A ℓ₁) (_≺_ : Rel A ℓ₂) : ∀ {m n} → REL (Vec A m) (Vec A n) (a ⊔ ℓ₁ ⊔ ℓ₂) where
```
but you can remove it elsewhere.